src/Uno.UI/UI/Xaml/ParentVisualTreeListener.cs

```diff
@@ -10,12 +10,15 @@ namespace Uno.UI.Xaml;
 internal class ParentVisualTreeListener
 {
 	private readonly DependencyObject _owner;
+	private readonly Action _onLoaded;
+	private readonly Action _onUnloaded;
 	private readonly SerialDisposable _registrations = new();
 
-	public ParentVisualTreeListener(DependencyObject owner)
+	public ParentVisualTreeListener(DependencyObject owner, Action onLoaded, Action onUnloaded)
 	{
 		_owner = owner;
-
+		_onLoaded = onLoaded;
+		_onUnloaded = onUnloaded;
```

Comment:
Could this cause memory leaks? We may need additional leak testing for this.

Reply:
Yes, will have to improve that still

Reply:
@jeromelaban is there a blocker for https://github.com/unoplatform/uno/pull/15540?